SPWwj/markbind
# Add QR Code Component

**What is the purpose of this pull request?**




**Overview of changes:**

This pull request introduces a new feature in our application, the `QrCodeProcessor` class. The `QrCodeProcessor` class processes custom QR code elements, replacing them with a div containing an SVG of the QR code and an optional caption. The attributes of the original element control various aspects of the QR code, including size, margin, error correction level, color, and background. 

**Anything you'd like to highlight/discuss:**

It would be worth discussing the default QR code options, such as size, margin, error correction level, color, and background, to ensure they meet our application's needs.

**Testing instructions:**
1. Use the `QrCodeProcessor` in a test file, with a variety of different node attributes to ensure all aspects of QR code generation are working as expected.
2. Test with and without optional caption attribute to ensure the caption functionality is working correctly.
3. Add the following QR code element in the MarkBind project and check the output:

```html
<qr-code 
    data="https://spirify.azurewebsites.net"
    size="10"
    margin="4"
    ec-level="H"
    color="red"
    background="white"
    type="svg"
    alt="Spirify QR Code"
    caption="Spirify QR Code"
    style="width:100%;height:100%;border:2px solid black;"
></qr-code>
```
4. View a live [demo of the QR Code feature](https://spwwj.github.io/mb-dev-SPWwj/contents/QrCode.html).

**Proposed commit message: (wrap lines at 72 characters)**

Add QrCodeProcessor Class

This commit adds a new feature to our application, the 
`QrCodeProcessor` class. This class processes custom QR code 
elements, replacing them with a div that contains an SVG of the QR 
code and an optional caption.

---

**Checklist:** :ballot_box_with_check:
